### PR TITLE
Fix build error and update to python3

### PR DESCRIPTION
**Pull request by pranith**
Not sure why it is complaining only now... seems like a very old issue.

g++ -o .opt_build/src/trace_read.o -c -O3 -std=c++14 -funroll-loops -Werror -Wuninitialized -Wno-write-strings -DLONG_COUNTERS -DNO_MPI -DNO_DEBUG -Isrc src/trace_read.cc
src/trace_read.cc: In constructor 'trace_reader_wrapper_c::trace_reader_wrapper_c()':
src/trace_read.cc:700:3: error: '*<unknown>.trace_reader_wrapper_c::m_dprint_output' is used uninitialized in this function [-Werror=uninitialized]
   m_dprint_output->close();
   ^~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
scons: *** [.opt_build/src/trace_read.o] Error 1

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python2
```
Comment (jiashenC) [inline build.py]: So we will still use python2 for running build script but use python3 for other config files?

Comment (pranith) [reply build.py]: Ideally, I should just do that change too. Good catch!

Comment (pranith) [reply build.py]: Updated with a new commit to change it to python3.

**Review by jiashenC — APPROVED** [review]
The Python part looks good to me.

**Pull request merged**